src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -69,126 +69,113 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
-            removePetServiceAccount(user, petServiceAccount).failed.foreach { e =>
+            removePetServiceAccount(petServiceAccount).failed.foreach { e =>
               logger.warn(s"Error occurred cleaning up pet service account [$petSaID] [$petSaDisplayName]", e)
             }
           }
         }
     }
   }
 
-  private def enablePetServiceAccount(userId: WorkbenchUserId, petServiceAccount: WorkbenchUserServiceAccount): Future[Unit] = {
+  private def enablePetServiceAccount(petServiceAccount: PetServiceAccount): Future[Unit] = {
     for {
-      _ <- directoryDAO.enableIdentity(petServiceAccount.subjectId)
-      _ <- googleDirectoryDAO.addMemberToGroup(WorkbenchGroupEmail(toProxyFromUser(userId.value)), petServiceAccount.email)
+      _ <- directoryDAO.enableIdentity(petServiceAccount.id)
+      _ <- googleDirectoryDAO.addMemberToGroup(WorkbenchEmail(toProxyFromUser(petServiceAccount.id.userId.value)), petServiceAccount.serviceAccount.email)
     } yield ()
   }
 
-  private def disablePetServiceAccount(userId: WorkbenchUserId, petServiceAccount: WorkbenchUserServiceAccount): Future[Unit] = {
+  private def disablePetServiceAccount(petServiceAccount: PetServiceAccount): Future[Unit] = {
     for {
-      _ <- directoryDAO.disableIdentity(petServiceAccount.subjectId)
-      _ <- googleDirectoryDAO.removeMemberFromGroup(WorkbenchGroupEmail(toProxyFromUser(userId.value)), petServiceAccount.email)
+      _ <- directoryDAO.disableIdentity(petServiceAccount.id)
+      _ <- googleDirectoryDAO.removeMemberFromGroup(WorkbenchEmail(toProxyFromUser(petServiceAccount.id.userId.value)), petServiceAccount.serviceAccount.email)
     } yield ()
   }
 
-  private def getPetServiceAccountForUser(userId: WorkbenchUserId): Future[Option[WorkbenchUserServiceAccount]] = {
-    directoryDAO.getPetServiceAccountForUser(userId).flatMap {
-      case Some(petEmail) => directoryDAO.loadSubjectFromEmail(petEmail.value).map {
-        case Some(petId: WorkbenchUserServiceAccountSubjectId) => Some(WorkbenchUserServiceAccount(petId, petEmail, WorkbenchUserServiceAccountDisplayName("")))
-        case _ => None
-      }
-      case None => Future.successful(None)
-    }
+  private def getPetServiceAccountsForUser(userId: WorkbenchUserId): Future[Seq[PetServiceAccount]] = {
+    directoryDAO.getAllPetServiceAccountsForUser(userId)
   }
 
-  private def setUpServiceAccount(user: WorkbenchUser, petServiceAccount: WorkbenchUserServiceAccount): Future[WorkbenchUserServiceAccountEmail] = {
+  private def setUpServiceAccount(petServiceAccount: PetServiceAccount): Future[PetServiceAccount] = {
     for {
-    // add Service Account User role to the configured emails so they can assume the identity of the pet service account
-      _ <- Future.traverse(petServiceAccountConfig.serviceAccountUsers) { email =>
-        googleIamDAO.addServiceAccountUserRoleForUser(petServiceAccountConfig.googleProject, petServiceAccount.email, email)
```

Comment:
@rtitle note this change - we are no longer adding ServiceAccountUser role. Instead we rely on the leo service account having project editor role at the org level.

Reply:
Yes; that makes sense.